qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/products/ProductsDbConnector.groovy

```diff
@@ -0,0 +1,126 @@
+package life.qbic.portal.qoffer2.products
+
+import groovy.sql.GroovyRowResult
+import groovy.util.logging.Log4j2
+import life.qbic.datamodel.dtos.business.services.DataStorage
+import life.qbic.datamodel.dtos.business.services.PrimaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.datamodel.dtos.business.services.ProductUnit
+import life.qbic.datamodel.dtos.business.services.ProductUnitFactory
+import life.qbic.datamodel.dtos.business.services.ProjectManagement
+import life.qbic.datamodel.dtos.business.services.SecondaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Sequencing
+import life.qbic.portal.portlet.exceptions.DatabaseQueryException
+import life.qbic.portal.portlet.packages.ListProductsDataSource
+import life.qbic.portal.qoffer2.database.ConnectionProvider
+import org.apache.groovy.sql.extensions.SqlExtensions
+
+import java.sql.ResultSet
+import java.sql.SQLException
+
+/**
+ * Provides a MariaDB connector implementation for {@link ListProductsDataSource}.
+ *
+ * @since 1.0.0
+ */
+@Log4j2
+class ProductsDbConnector implements ListProductsDataSource {
+
+  private final ConnectionProvider provider
+
+  /**
+   * Creates a connector for a MariaDB instance.
+   *
+   * The class instantiation will fail, if the passed provider is null.
+   *
+   * @param provider A connection provider
+   */
+  ProductsDbConnector(ConnectionProvider provider) {
+    this.provider = Objects.requireNonNull(provider, "Provider must not be null.")
+  }
+
+  @Override
+  List<Product> findAllAvailableProducts() throws DatabaseQueryException {
+    try {
+      tryToFindAllProducts()
+    } catch (SQLException e) {
+      log.error(e.message)
+      log.error(e.stackTrace.join("\n"))
+      throw new DatabaseQueryException("Unable to list all available products.")
+    }
+  }
+
+  private List<Product> tryToFindAllProducts() {
```

Comment:
```suggestion
  private List<Product> findAllProducts() {
```

I think it would suffice to name it like this

Reply:
I applied bob martin's clean code :)